Cognite.Testing/TestUtils.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Cognite.Extractor.Testing
+{
+    /// <summary>
+    /// Collection of static utility methods for testing
+    /// </summary>
+    public static class TestUtils
+    {
+        /// <summary>
+        /// Returns a task that polls an asynchronous method every 200 milliseconds and returns once it
+        /// is true. This throws an exception if the condition does not become true within <paramref name="seconds"/>.
+        /// </summary>
+        /// <param name="condition">Method to poll</param>
+        /// <param name="seconds">Number of seconds to wait before failing</param>
+        /// <param name="assertion">Method returning a string to include in the thrown exception</param>
+        /// <exception cref="ArgumentNullException">if condition or assertion are null</exception>
+        /// <exception cref="TrueException">
+        /// If <paramref name="condition"/> does not become true within <paramref name="seconds"/>
+        /// </exception>
+        public static async Task WaitForCondition(Func<Task<bool>> condition, int seconds, Func<string> assertion)
+        {
+            if (condition == null) throw new ArgumentNullException(nameof(condition));
```

Comment:
I'm not sure what exists in C#, but is there something like [lombok](https://projectlombok.org/features/NonNull)? We're using this type of decorators to do null checks in extration pipelines quite a bit and it's very nice, and way shorter than writing it out every time. Worth investigating at least

Reply:
For reference it is default. I.e. the assumption `condition is not null` is present here, but it only generates a warning if the calling project also has enabled nullable analysis.